### PR TITLE
sync: RwLock add try_read and try_write methods

### LINKED ISSUE #2284: RwLock miss try_read and try_write method
## Version 0.2.13

> ```
> └── tokio v0.2.13
>     └── tokio-macros v0.2.5
> ```

## Platform

archlinux x64

## Description

I try 
```rust
use tokio::sync::RwLock;

#[tokio::main]
async fn main() {
    let rw_lock = RwLock::new(1);
    
    rw_lock.try_read();
}
```

It should compile well. But cargo report

> method not found in `tokio::sync::rwlock::RwLock<{integer}>`

I read the doc, RwLock doesn't have `try_read` or `try_write`, if I want to try to acquire a read lock or write lock but don't block when acquire failed, what should I do

### PR DESCRIPTION
**Pull request by nylonicious**
Closes #2284

### CODE REVIEW

File: tokio/src/sync/rwlock.rs [hunk 1]
```diff
@@ -441,6 +441,52 @@ impl<T: ?Sized> RwLock<T> {
...
+    ///         // While main has an active read lock, we acquire one too.
+    ///         let n = c_lock.read().await;
+    ///         assert_eq!(*n, 1);
+    ///     }).await.expect("The spawned task has panicked");
+    ///
+    ///     // Drop the guard when spawned task finishes.
+    ///     drop(v);
+    /// }
+    /// ```
+    pub fn try_read(&self) -> Result<RwLockReadGuard<'_, T>, TryReadError> {
```
Comment (carllerche) [inline tokio/src/sync/rwlock.rs:476]: I don't think we need to create new error types for each fn here. We could reuse `TryLockError`. This matches what `std` does.

**Review by carllerche — CHANGES_REQUESTED** [review]
Thanks for doing this. The implementations look good to me. I left a note about the error types.

**Review by carllerche — APPROVED** [review]
Thanks :+1:

File: tokio/src/sync/rwlock.rs [hunk 2]
```diff
@@ -441,6 +442,52 @@ impl<T: ?Sized> RwLock<T> {
         }
     }
 
+    /// Attempts to acquire this `RwLock` with shared read access.
+    ///
+    /// If the access couldn't be acquired immediately returns [`TryLockError`].
```
Comment (hawkw) [inline tokio/src/sync/rwlock.rs:447]:
```suggestion
    /// If the access couldn't be acquired immediately, returns [`TryLockError`].
```

File: tokio/src/sync/rwlock.rs [hunk 3]
```diff
@@ -476,6 +523,43 @@ impl<T: ?Sized> RwLock<T> {
         }
     }
 
+    /// Attempts to acquire this `RwLock` with exclusive write access.
+    ///
+    /// If the access couldn't be acquired immediately returns [`TryLockError`].
```
Comment (hawkw) [inline tokio/src/sync/rwlock.rs:528]:
```suggestion
    /// If the access couldn't be acquired immediately, returns [`TryLockError`].
```

**Review by hawkw — APPROVED** [review]
looks good to me — one small docs nit

**Pull request merged**